cmd/lingo/main.go

```diff
@@ -153,7 +153,8 @@ func run() error {
 	go autoscaler.Start()
 
 	proxy.MustRegister(metricsRegistry)
-	proxyHandler := proxy.NewHandler(deploymentManager, endpointManager, queueManager)
+	var proxyHandler http.Handler = proxy.NewHandler(deploymentManager, endpointManager, queueManager)
+	proxyHandler = proxy.NewRetryMiddleware(3, proxyHandler)
```

Comment:
todo: hard coded 3 retries but it would be better to have a start param to make this configurable. `0` would disable the middleware

Reply:
Sounds like a good configurable option. I am fine with hardcoding everything in this PR and adding another PR for configuration.